mesa/experimental/cell_space/grid.py

```diff
@@ -105,7 +106,103 @@ def _connect_single_cell_2d(self, cell: T, offsets: list[tuple[int, int]]) -> No
                 cell.connect(self._cells[ni, nj])
 
 
-class OrthogonalMooreGrid(Grid[T]):
+class _PropertyGrid(Grid, Generic[T]):
+    """
+    A private subclass of Grid that supports the addition of property layers, enabling
+    the representation and manipulation of additional data layers on the grid. This class is
+    intended for internal use within the Mesa framework.
+
+    The `_PropertyGrid` extends the capabilities of a basic grid by allowing each cell
+    to have multiple properties, each represented by a separate PropertyLayer.
+    These properties can be used to model complex environments where each cell
+    has multiple attributes or states.
+
+    Attributes:
+        properties (dict): A dictionary mapping property layer names to PropertyLayer instances.
+
+    Methods:
+        add_property_layer(property_layer): Adds a new property layer to the grid.
+        remove_property_layer(property_name): Removes a property layer from the grid by its name.
+        select_random_empty_cell(self): Returns an empty random cell instance from the grid.
+
+
+    Note:
+        This class is not intended for direct use in user models but is currently used by the OrthogonalMooreGrid and OrthogonalVonNeumannGrid.
+    """
+
+    def __init__(
+        self,
+        dimensions: Sequence[int],
+        torus: bool = False,
+        capacity: float | None = None,
+        random: Random | None = None,
+        cell_klass: type[T] = Cell,
```

Comment:
I checked that the existing implementation uses `cell_klass`, but shouldn't this be `cell_class`?

Reply:
In the parent grid class, it was referred to as cell_klass. I assumed that was the correct convention.

Reply:
Yes there is no clear convention, but I think the problem these names all try to solve is when you want to name a variable class directly, which is forbidden in Python since `class` is a reserved keyword. I think here it would be no problem and probably clerarer to just name it `cell_class`. But no worries @haleelsada it was correct to just use the same naming as the parent. But +1 from my side for changing it in a separate PR.